File: Missing values/missing values.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Sun Jan 28 00:41:28 2018

@author: aalvarez
"""
##Librerias
import pandas as pd 
import numpy as np 

#Leer el dataset
dataset = pd.read_csv('prestamos.csv')

#Ver valores nulos
print('Valores nulos:\n', dataset.isnull().sum())


#1.Eliminar valores nulos
dataset = dataset.loc[dataset['Sexo'].isnull() == False]
dataset = dataset.loc[dataset['CantidadPrestamo'].isnull() == False]
dataset = dataset.loc[dataset['Plazo'].isnull() == False]
dataset = dataset.loc[dataset['Casado'].isnull() == False]

#2.Inputar valores nulos
dataset['Sexo'].fillna(dataset['Sexo'].mode()[0],inplace=True)
dataset['CantidadPrestamo'].fillna(dataset['CantidadPrestamo'].mean(),inplace=True)
dataset['Plazo'].fillna(dataset['Plazo'].mean(),inplace=True)
dataset['Casado'].fillna(dataset['Casado'].mode()[0],inplace=True)

File: Missing values/missing values.r
#Leer el dataset
dataset = read.csv('prestamos.csv')

#Ver valores nulos
summary(dataset)

#0.Decodificar los NA
dataset$Sexo[dataset$Sexo == ""] = NA
dataset$Casado[dataset$Casado == ""] = NA


#1.Eliminar valores nulos
dataset=na.omit(dataset)


#2.Inputar valores nulos
dataset$Sexo = ifelse(is.na(dataset$Sexo), as.character(mode(dataset$Sexo)), as.character(dataset$Sexo))
dataset$Casado = ifelse(is.na(dataset$Casado), as.character(mode(dataset$Casado)), as.character(dataset$Casado))
dataset$CantidadPrestamo = ifelse(is.na(dataset$CantidadPrestamo), mean(dataset$CantidadPrestamo, na.rm=TRUE), dataset$CantidadPrestamo)
dataset$Plazo = ifelse(is.na(dataset$Plazo), mean(dataset$Plazo, na.rm=TRUE), dataset$Plazo)

